### PR TITLE
Update versions

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://bitbucket.org/al45tair/netifaces/issues/53/update-supported-python-versions.

Assumes support for the currently supported CPython versions (2.7, 3.4+).

https://en.wikipedia.org/wiki/CPython#Version_history

Here's the pip installs for netifaces from PyPI for last month, which show the unsupported versions aren't used much:

| python_version | percent | download_count |
| -------------- | ------: | -------------: |
| 2.7            |  82.89% |        358,329 |
| 3.6            |   7.44% |         32,167 |
| 3.5            |   6.40% |         27,675 |
| 3.4            |   3.00% |         12,965 |
| 2.6            |   0.22% |            972 |
| 3.7            |   0.03% |            135 |
| 3.3            |   0.01% |             46 |
| 3.2            |   0.00% |              3 |

Source: `pypinfo --start-date -56 --end-date -26 --percent --pip --markdown netifaces pyversion`

### CODE REVIEW

**Comment by johnthagen** [comment]
Travis also has support for [PyPy](https://pypy.org/), could consider adding support for it.

**Comment by hugovk** [comment]
Added!

File: netifaces.c [hunk 1]
```diff
@@ -1,18 +1,5 @@
...
-{
-  Py_ssize_t len = strlen(s);
-  if (!len) {
-    Py_UNICODE uc = 0;
-    return PyUnicode_FromUnicode(&uc, 0);
-  }
-  return PyUnicode_DecodeUTF8(s, len, NULL);
-}
-#endif
-
```
Comment (al45tair) [inline netifaces.c]: I don't see a good reason for removing this code. I agree that few people are probably using Python 2.5 or earlier any more, but this doesn't break anything for later versions.

**Comment by al45tair** [comment]
I've added the extra versions to the testing matrix, as I think that's sensible. I don't think it's a good idea to remove the Python <=2.5 compatibility code, as I can't see any benefit from doing that (it's only a short snippet, and taking it out will just result in breakage if someone tries to install on Python 2.5; yes, I know, not many people will, but there's no reason to break things for them at this point).

**Pull request closed without merging**